client/src/test/java/io/pravega/schemaregistry/client/SchemaRegistryClientAuthTest.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.client;
+
+import com.google.common.collect.Lists;
+import io.pravega.schemaregistry.common.CredentialProvider;
+import io.pravega.test.common.AssertExtensions;
+import org.junit.Test;
+
+import javax.ws.rs.ProcessingException;
+import java.net.URI;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.junit.Assert.assertEquals;
+
+public class SchemaRegistryClientAuthTest {
+    
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testGroup() {
+        URI schemaRegistryUri = URI.create("http://localhost:9092");
+
+        AtomicLong methodCounter = new AtomicLong(0);
+        AtomicLong tokenCounter = new AtomicLong(0);
+        SchemaRegistryClientConfig config = SchemaRegistryClientConfig.builder()
+                                                                      .authentication(new CredentialProvider() {
+                                                                          @Override
+                                                                          public String getMethod() {
+                                                                              methodCounter.incrementAndGet();
+                                                                              return "MyMethod";
+                                                                          }
+
+                                                                          @Override
+                                                                          public String getToken() {
+                                                                              tokenCounter.getAndIncrement();
```

Comment:
Curios, is there a reason for using `incrementAndGet()` above and `getAndIncrement()` here. I guess the effect'd be the same in this case. The juxtaposition draws some attention, and they's why the question. Maybe you should modify it to use the same atomic method, to avoid the juxtaposition .